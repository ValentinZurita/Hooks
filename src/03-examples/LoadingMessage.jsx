import React from 'react'

export const LoadingMessage = () => {
  // Return a loading spinner
  return (
    <div className=" text-center mt-5 mb-5">
  
      <div className="spinner-border" role="status">
     
      </div>
    </div>
  )
}
