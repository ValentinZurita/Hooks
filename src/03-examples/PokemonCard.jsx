import React from 'react'
import PropTypes from 'prop-types'

export const PokemonCard = ({ id, name, sprites = [] }) => {
  // Return the pokemon card
  return (
    // Carta del pokemon
    <section className="card m-2 p-2" style={{ width: '500px' }}>
      {/* Sprites del pokemon */}
      <div className="d-flex justify-content-center flex-wrap">
        {sprites.map((sprite) => (
          <img
            key={sprite}
            src={sprite}
            className="card-img-top mx-1"
            alt={name}
            style={{ width: '200px', height: '200px' }}
          />
        ))}
      </div>

      {/* Nombre del pokemon */}
      <div className="card-body">
        <h4 className="text-center text-capitalize">
          #{id} {name}
        </h4>
      </div>
    </section>
  )
}

// Definici√≥n de las propiedades del componente
PokemonCard.propTypes = {
  sprites: PropTypes.object,
  id: PropTypes.number,
  name: PropTypes.string,
}
