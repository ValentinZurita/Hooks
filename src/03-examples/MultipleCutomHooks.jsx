import React from 'react'
import { useFetch } from '../hooks/useFetch'
import { useCounter } from '../hooks/useCounter'
import { PokemonCard } from './PokemonCard'
import { LoadingMessage } from './LoadingMessage'

export const MultipleCutomHooks = () => {
  const { counter, decrement, increment } = useCounter(1)
  const { data, isLoading, hasError } = useFetch(
    `https://pokeapi.co/api/v2/pokemon/${counter}`
  )

  return (
    <>
      {/*Título */}
      <h1 className='text-center'>Información Pokémon:</h1>
      <hr />

      {/*Mostrar Pokemon */}
      {console.log(isLoading)}
      <div className="d-flex justify-content-center">
        {isLoading ? (
          <LoadingMessage />
        ) : (
          <PokemonCard
            sprites={[
              data?.sprites?.front_default,
              data?.sprites?.back_default,
            ]}
            id={counter}
            name={data?.name}
          />
        )}
      </div>

      {/*Botones para cambiar pokemones */}
      <div className="d-flex justify-content-center">
        <button
          className="btn btn-primary mt-2"
          onClick={() => (counter > 1 ? decrement(1) : null)}
        >
          Anterior
        </button>
        <button
          className="btn btn-primary mt-2"
          onClick={() => (counter < 898 ? increment(1) : null)}
        >
          Siguiente
        </button>
      </div>
    </>
  )
}
