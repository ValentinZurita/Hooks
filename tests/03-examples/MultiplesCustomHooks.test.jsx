import React from 'react'
import { render } from '@testing-library/react'
import { MultipleCutomHooks } from '../../src/03-examples/MultipleCutomHooks'

describe('Pruebas del componente MultiplesCustomHooks', () => {
  test('Debe hacer un snapshot', () => {
    const { asFragment } = render(<MultipleCutomHooks />)
    expect(asFragment()).toMatchSnapshot()
  })

  

})
