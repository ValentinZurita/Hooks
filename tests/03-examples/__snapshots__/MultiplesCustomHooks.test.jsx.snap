// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente MultiplesCustomHooks Debe hacer un snapshot 1`] = `
<DocumentFragment>
  <h1
    class="text-center"
  >
    Información Pokémon:
  </h1>
  <hr />
  <div
    class="d-flex justify-content-center"
  >
    <div
      class=" text-center mt-5 mb-5"
    >
      <div
        class="spinner-border"
        role="status"
      />
    </div>
  </div>
  <div
    class="d-flex justify-content-center"
  >
    <button
      class="btn btn-primary mt-2"
    >
      Anterior
    </button>
    <button
      class="btn btn-primary mt-2"
    >
      Siguiente
    </button>
  </div>
</DocumentFragment>
`;
